jax/_src/nn/initializers.py

```diff
@@ -129,15 +130,15 @@ def _compute_fans(shape: core.NamedShape, in_axis=-2, out_axis=-1,
   if isinstance(in_axis, int):
     in_size = shape[in_axis]
   else:
-    in_size = int(np.prod([shape[i] for i in in_axis]))
+    in_size = operator.index(prod(shape[i] for i in in_axis))
   if isinstance(out_axis, int):
     out_size = shape[out_axis]
   else:
-    out_size = int(np.prod([shape[i] for i in out_axis]))
+    out_size = operator.index(prod(shape[i] for i in out_axis))
   if isinstance(batch_axis, int):
     batch_size = shape[batch_axis]
   else:
-    batch_size = int(np.prod([shape[i] for i in batch_axis]))
+    batch_size = operator.index(prod(shape[i] for i in batch_axis))
```

Comment:
`np.prod([])` will return `1.0` of type `np.float64`, and `prod` is faster than `np.prod` in most cases.